### ISSUE TITLE
Is it needed to make sure  CRDs exist before starting the controller ?

### ISSUE THREAD

**Issue by zhucan** [pingcap/tidb-operator#3939]
https://github.com/pingcap/tidb-operator/blob/c76d162b88fa02047193c19480cf5e1044d50f34/cmd/controller-manager/main.go#L183

**Comment by DanielZhangQD** [comment]
@zhucan It's a good idea, would you like to submit PR to improve this?

**Comment by zhucan** [comment]
@DanielZhangQD yeah, I had tested, but I have a question.  When running tidb-controller,  it's need to start six controller, but when I exec "kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/v1.1.12/manifests/crd.yaml" , the "dmclusters.pingcap.com"  not installed. 
![image](https://user-images.githubusercontent.com/13997135/116025695-b571e900-a683-11eb-96de-83766af24be0.png)

**Comment by zhucan** [comment]
The installed crd not match to the needed crd to start the controller.

**Comment by DanielZhangQD** [comment]
DM cluster is only available in the master branch.

**Comment by DanielZhangQD** [comment]
BTW, `Is it needed to make sure CRDs exist before starting the controller` is this really necessary?
For the current process, we request the user to install the CRD first in the doc, and if users missed that step, logs will be printed in the log of tidb-controller-manager.
Even if we add a check before the controllers start, we still print logs, right?

**Comment by zhucan** [comment]
yes, still print logs, but the pod(controller-manager) will not in a healthy status, it's no need to runing, the custom will know that missed install crd.

**Comment by DanielZhangQD** [comment]
OK, could you please submit the PR?
